tests/validation/tests/v3_api/test_registry.py

```diff
@@ -126,38 +396,40 @@ def delete_registry(client, registry, ns):
     print("Project ID")
     print(project.id)
     client.delete(registry)
-    registryname = registry.name
-    # Sleep to allow for the registry to be deleted
     time.sleep(5)
-    print("Registry list after deleting registry")
-    registrydict = client.list_dockerCredential(name=registryname).data
-    print(registrydict)
-    if len(registrydict) == 0:
-        assert True
-
-    namespacedict = c_client.list_namespace(projectId=project.id).data
-    print("List of namespaces")
-    print(namespacedict)
-    len_namespace = len(namespacedict)
-    namespaceData = namespacedict
-
-    # Registry is essentially a secret, deleting the registry should delete
-    # the secret. Verify secret is deleted by "kubectl get secret" command
-    # for each of the namespaces
-    for i in range(0, len_namespace):
-        ns_name = namespaceData[i]['name']
-        print(i, ns_name)
-
-        command = " get secret " + registryname + " --namespace=" + ns_name
-        print("Command to obtain the secret")
-        print(command)
-        result = execute_kubectl_cmd(command, json_out=False, stderr=True)
-        print(result)
-
-        print("Verify that the secret does not exist "
-              "and the error code returned is non zero ")
-        if result != 0:
-            assert True
+    try:
+        registryname = registry.name
+    except AttributeError:
+        registryname = None
+    # Sleep to allow for the registry to be deleted
+    if registryname is not None:
+        print("Registry list after deleting registry")
+        registrydict = client.list_dockerCredential(name=registryname).data
+        print(registrydict)
+        assert len(registrydict) == 0, "Unable to delete registry"
+
+        namespacedict = c_client.list_namespace(projectId=project.id).data
+        print("List of namespaces")
+        print(namespacedict)
+        len_namespace = len(namespacedict)
+        namespaceData = namespacedict
+
+        # Registry is essentially a secret, deleting the registry should delete
+        # the secret. Verify secret is deleted by "kubectl get secret" command
+        # for each of the namespaces
+        for i in range(0, len_namespace):
+            ns_name = namespaceData[i]['name']
+            print(i, ns_name)
+
+            command = " get secret " + registryname + " --namespace=" + ns_name
+            print("Command to obtain the secret")
+            print(command)
+            result = execute_kubectl_cmd(command, json_out=False, stderr=True)
+            print(result)
+
+            print("Verify that the secret does not exist "
+                  "and the error code returned is non zero ")
+            assert result != 0, "Error code is 0!"
```

Comment:
The verification for the secret deletion should be included always, not only when registryname is not None.

Reply:
Done